recipes/jfalcou-eve/all/conanfile.py

```diff
@@ -40,6 +40,8 @@ def validate(self):
             tools.check_min_cppstd(self, self._min_cppstd)
         if self.settings.compiler == "Visual Studio":
             raise ConanInvalidConfiguration("EVE does not support MSVC yet (https://github.com/jfalcou/eve/issues/1022).")
+        if self.settings.compiler == "apple-clang":
+            raise ConanInvalidConfiguration("EVE does not support apple Clang due to an incomplete libcpp.")
```

Comment:
Same here. If `v2021.10.0` was building before, why removing it now? If only `v2022.03.0` is known not to work with `apple-clang` just raise for that version.

Reply:
`apple-clang` was not available when the first recipe was written. No version of EVE ever worked on Apple Clang as the Apple Clang libcpp doesn't contain half of what we require.

Reply:
It was not discarded before, right? The CI runs the build for all the configurations (Windows, Macos, Linux) and will only skip those that raise `ConanInvalidConfiguration`.

Reply:
I'm a bit lost here. The old version never passed CI with apple-clang as it predates its support in the index.

Reply:
`apple-clang` has been available in CCI from the very beginning.

It is a header-only, so it was available for all configurations: 

```python
    def package_id(self):
        self.info.header_only()
```

And `test_package/conanfile.py` works with `apple-clang` (I've just tried it locally on my machine).

Reply:
I'm utterly confused then. Sorry for all the noise but it looked like the first issue with this PR was that it failed on apple-clang.

See previous message:
```
Users/jenkins/w/prod/BuildSingleReference@2/.conan/data/jfalcou-eve/v2021.10.0/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/eve/concept/vectorized.hpp:46:62: error: no member named 'integral' in namespace 'std'; did you mean 'internal'?
  concept integral_simd_value        = simd_value<T> && std::integral<detail::value_type_t<T>>;
```

and also, you said earlier:

```
The latest PR to this recipe is 5 months old, by that time apple-clang 13 was not used here. So this is the first time this recipe tries this compiler
``` 

So I don't see how it can work on your machine unless your apple-clang is not the CI apple-clang ?
Again sorry for this mess. I have no access to an Apple machine so I can't actually test those.

Reply:
I'm already using apple-clang 13.1 (default for the latest MacOS version), maybe it defaults to a different C++ standard where that member is defined. [CI uses](https://github.com/conan-io/conan-center-index/blob/master/docs/supported_platforms_and_configurations.md#macos) 11.0.3, 12.0.5 and 13.0.0

Anyway, if you know it shouldn't work for `apple-clang`, raise `ConanInvalidConfiguration` for that compiler. We can enable it later in some other PR.

Reply:
> `apple-clang` was not available when the first recipe was written. No version of EVE ever worked on Apple Clang as the Apple Clang libcpp doesn't contain half of what we require.

does it work with the gcc implementation `libstdc++` instead of the native one `libc++`? just curious that usually the work around